spec/acceptance/foreman_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'spec_helper_acceptance'
+
+describe 'certs::foreman' do
+  FQDN = fact('fqdn')
+
+  before(:all) do
+    manifest = <<~MANIFEST
+      file { '/etc/foreman':
+        ensure => directory,
+      }
+
+      group { 'foreman':
+        ensure => present,
+        system => true,
+      }
+    MANIFEST
+    apply_manifest(manifest, catch_failures: true)
+  end
+
+  context 'with default parameters' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) { 'include certs::foreman' }
+    end
+
+    describe x509_certificate('/etc/foreman/client_cert.pem') do
+      it { should be_certificate }
+      it { should be_valid }
+      it { should have_purpose 'client' }
+      include_examples 'certificate issuer', "C = US, ST = North Carolina, L = Raleigh, O = Katello, OU = SomeOrgUnit, CN = #{FQDN}"
```

Comment:
I'm seeing this multiple times. That says to me there is an opportunity for de-duplication.

Reply:
The thing is that they sometimes do differ a bit. However, we're talking about completely redesigning this so it doesn't really make sense to change it now.